### PR TITLE
Feature/determine default branch

### LINKED ISSUE #118: Determine default branch for repository
Default branch for new repositories in Github is called `main`. Because of that, we should determine which branch is default for repository before we land changes.

Possible solution:
- get default branch from git;
- get default branch name from phabricator;
- keep value in arcrc;

### PR DESCRIPTION
**Pull request by szewczykmira**
Closes #118

### CODE REVIEW

File: phlq/pharbicator/api.py [hunk 1]
```diff
@@ -37,12 +37,21 @@ def send(self, query_path: str, payload_info: Optional[dict] = None) -> dict:
...
-    def repository_search(self, repository_callsign: str):
+    def repository_search(
+        self,
+        repository_callsign: Optional[str] = None,
+        repository_url: Optional[str] = None,
+    ):
         query_path = "diffusion.repository.search"
         data = {
-            "constraints[callsigns][]": [repository_callsign],
+            # "constraints[callsigns][]": [repository_callsign],
```
Comment (kam193) [inline phlq/pharbicator/api.py:47]: Is here any idea to keep this commented?

File: phlq/pharbicator/utils.py [hunk 2]
```diff
@@ -17,3 +21,12 @@ def validate_git_protocol(repo_url: str) -> bool:
...
     return repo_url.startswith(protocol_preffixes)
+
+
+def get_default_branch_name(
+    phabricator_client: "PhabricatorAPIClient", repository_url: str
+) -> str:
+    repository_data = phabricator_client.repository_search(
+        repository_url=repository_url
+    )
+    return repository_data.get("fields").get("defaultBranch")
```
Comment (kam193) [inline phlq/pharbicator/utils.py:32]: Do we accept `AttributeError` here? Maybe better use `repository_data.get("fields", {}).get("defaultBranch")`

**Review by kam193 — CHANGES_REQUESTED** [review]

**Review by bart84ek — APPROVED** [review]

File: phlq/pharbicator/api.py (hunk 1 above)
Comment (szewczykmira) [reply phlq/pharbicator/api.py:47]: Naaah, this should be deleted :)

File: phlq/pharbicator/utils.py (hunk 2 above)
Comment (szewczykmira) [reply phlq/pharbicator/utils.py:32]: Yeah, `repository_data.get("fields", {}).get("defaultBranch")` propably was what I was going to use, but forget to do it :P

Comment (kam193) [reply phlq/pharbicator/utils.py:32]: And maybe add some small test...? :D

Comment (szewczykmira) [reply phlq/pharbicator/utils.py]: @kam193 test added

File: tests/phabricator/test_utils.py [hunk 3]
```diff
@@ -46,3 +48,18 @@ def test_get_repository_path(url, repository_path):
...
+        ({"fields": {"defaultBranch": "master"}}, "master"),
+        ({"fields": {"defaultBranch": "main"}}, "main"),
+        ({"fields": {"default": "main"}}, None),
+        ({}, None),
+    ],
+)
+def test_get_default_branch_name(repository_data, output):
+    phabricator_client = Mock()
+    phabricator_client.repository_search = Mock(return_value=repository_data)
+    assert get_default_branch_name(phabricator_client, "") == output
```
Comment (kam193) [inline tests/phabricator/test_utils.py:65]: If we mock phabricator, maybe add also one line more and check if we called it properly? ;)

**Review by kam193 — APPROVED** [review]

File: tests/phabricator/test_utils.py (hunk 3 above)
Comment (szewczykmira) [reply tests/phabricator/test_utils.py:65]: I'm not sure that this is something we need to test (phabricator api is tested with vcr and we are just passing argument from higher function), but if you want it ...

**Pull request merged**